rocon_gateway/src/rocon_gateway/master_api.py

```diff
@@ -43,6 +48,24 @@ class LocalMaster(rosgraph.Master):
     def __init__(self):
         rosgraph.Master.__init__(self, rospy.get_name())
 
+        self.connections_lock = threading.Lock()
+        self.connections = utils.create_empty_connection_type_dictionary(set)
+        # in case this class is used directly (script call) we need to find the connection cache
+
+        connection_cache_namespace = rocon_gateway_utils.resolve_connection_cache(rospy.Time(5))
```

Comment:
Need to background this lookup so it isn't going to fall over on an indeterminate timeout.